ansible/roles/debops.postldap/templates/etc/ansible/facts.d/postldap.fact.j2

```diff
@@ -0,0 +1,12 @@
+#!{{ ansible_user_shell|d("/usr/bin/env sh") }}
+
+cat << EOF
+{
+    "installed": true,
+    "vmail_posix_user" : "{{ postldap__vmail_posix_user }}",
+    "vmail_posix_uidnumber" : {{ postldap__register_vmail_posix_user.uid|int }},
+    "vmail_posix_group" : "{{ postldap__vmail_posix_group }}",
+    "vmail_posix_gidnumber" : "{{ postldap__register_vmail_posix_user.group|int }}",
+    "mailbox_base" : "{{ postldap__mailbox_base }}"
+}
+EOF
```

Comment:
Good first try, but you should use Python instead of Bash for the fact scripts. There are lots of good examples in various roles.

Optimally the facts should get the information themselves, not rely on Jinja templating - that should only be for data that cannot be inferred from the host itself, for example if particular option is defined in the Ansible role. In this particular case, you could check if some Dovecot binary is present to define if Dovecot is installed, store the user and group name in the fact script via Jinja and JSON loading, and get their corresponding UID and GID from the system user and password database via the `pwd` and `grp` Python modules. I'm not sure about mailbox base - do we go with the home directory of the `vmail` user, or perhaps a subdirectory? I guess that because this directory is created by the role itself, it should be stored in the script via Jinja and not inferred somehow from the host since I wouldn't know from where get that information.